source/agora/consensus/protocol/Nominator.d

```diff
@@ -556,6 +563,61 @@ extern(D):
         this.active_timers[type] = this.taskman.setTimer(
             timeout.msecs, { callCPPDelegate(callback); });
     }
+
+    /***************************************************************************
+
+        Used by the nomination protocol to randomize the order of messages
+        between nodes.
+
+        Params:
+            slot_idx = the slot index we're currently reaching consensus for.
+            prev = the previous data set for the provided slot index.
+            is_priority = the flag to check that this call is for priority.
+            round_number = the nomination round
+            node_id = the id of the node for which this computation is being made
+
+    ***************************************************************************/
+
+    public override uint64_t computeHashNode(uint64_t slot_idx,
+        ref const(Value) prev, bool is_priority, int32_t round_number,
+        ref const(NodeID) node_id) nothrow
+    {
+        uint512 hash;
+        const random_seed = this.ledger.getValidatorRandomSeed(Height(slot_idx));
```

Comment:
You're supposed to use the previous height, not the current one..

Reply:
Lets consider the case where block height 1 was just externalized. We will have at least the preimages for height 1, but there is no guarantee we have preimages for heights 2+.

So when we start nominating block height 2, we can only take the most recent guaranteed preimages - the ones for height 1 

So you just have to do `slot_idx - 1` in this call.

Reply:
> So when we start nominating block height 2, we can only take the most recent guaranteed preimages - the ones for height 1

I got it. Thanks.